packages/discussions/src/types.ts

```diff
@@ -836,3 +852,7 @@ export interface IDiscussionsMentionMeta {
   postId: string;
   replyId?: string;
 }
+
+export interface IDiscussionsUser extends IUser {
+  username: string | undefined | null;
```

Comment:
I believe the way this is written, it requires a `username` property be defined, but it can have values of type `string` `undefined` or `null`. The [IUser](https://github.com/Esri/arcgis-rest-js/blob/main/packages/arcgis-rest-request/src/types/user.ts#L17) that we extend defines this property as `username?: string;` which I believe means the `username` property may not be defined. Maybe we should consider rewriting this as `username?: string | null;`?

Reply:
Changed in https://github.com/Esri/hub.js/pull/881/commits/b4beb938c6245ebb33f9225a627b4339ecafa499.